python/src/main.cpp

```diff
@@ -106,23 +124,41 @@ PYBIND11_MODULE(kp, m) {
             "Records operation to sync tensor(s) from GPU memory to local memory using staging tensors")
         .def("record_algo_mult", &kp::Sequence::record<kp::OpMult>,
             "Records operation to run multiplication compute shader to two input tensors and an output tensor")
-        .def("record_algo_file", &kp::Sequence::record<kp::OpAlgoBase, std::string>,
-            "Records an operation using a custom shader provided from a shader path")
+        .def("record_algo_file", [](kp::Sequence &self, 
+                                    std::vector<std::shared_ptr<kp::Tensor>> tensors,
+                                    const std::string& file_path,
+                                    std::tuple<uint32_t,uint32_t,uint32_t> work_group) -> bool {
+                const kp::OpAlgoBase::KomputeWorkgroup wgroup{
+                    std::get<0>(work_group), std::get<1>(work_group), std::get<2>(work_group),
+                };
+                return self.record<kp::OpAlgoBase>(tensors, file_path, wgroup);
+            },
+            "Records an operation using a custom shader provided from a shader path",
+            py::arg("tensors"), py::arg("file_path"), py::arg("work_group") = std::make_tuple(0,0,0)  )
         .def("record_algo_data", [](kp::Sequence &self,
                                     std::vector<std::shared_ptr<kp::Tensor>> tensors,
-                                    py::bytes &bytes) -> float {
+                                    py::bytes &bytes,
+                                    std::tuple<uint32_t,uint32_t,uint32_t> work_group) -> bool {
```

Comment:
Probably best for this one to be optional param (with defaults)

Reply:
`work_group` is already an optional argument, see below in line 154: `py::arg("work_group") = std::make_tuple(0,0,0)`